test/integration/idle_timeout_integration_test.cc

```diff
@@ -189,6 +194,23 @@ TEST_P(IdleTimeoutIntegrationTest, PerStreamIdleTimeoutHeadRequestAfterDownstrea
   EXPECT_EQ("", response->body());
 }
 
+TEST_P(IdleTimeoutIntegrationTest,
+       PerStreamIdleTimeoutHeadRequestAfterDownstreamJsonContentTypeRequest) {
+  enable_per_stream_idle_timeout_ = true;
+  auto response = setupPerStreamIdleTimeoutTest("GET", "application/json");
+
+  waitForTimeout(*response, "downstream_rq_idle_timeout");
+  EXPECT_FALSE(upstream_request_->complete());
+  EXPECT_EQ(0U, upstream_request_->bodyLength());
+  EXPECT_TRUE(response->complete());
+  EXPECT_STREQ("408", response->headers().Status()->value().c_str());
+  EXPECT_EQ(Http::Headers::get().ContentTypeValues.Json,
+            response->headers().ContentType()->value().c_str());
+  envoy::data::core::v2alpha::LocalReply local_reply;
+  local_reply.set_body("stream timeout");
+  EXPECT_EQ(MessageUtil::getJsonStringFromMessage(local_reply, true, true), response->body());
```

Comment:
This may not always produce deterministic equality. A better approach is to take the returned JSON and load it into a local_reply message, and then use TestUtility::ProtoEq